Baboon_stuff/Baboon_Reader.cpp

```diff
@@ -1,77 +1,152 @@
-// reading a text file
 #include <iostream>
 #include <fstream>
 #include <string>
 #include <vector>
 #include <sstream>
 #include <algorithm>
 #include <map>
+#include <utility>      // std::pair, std::make_pair
+#include <bits/stdc++.h>
 using namespace std;
-/*store timestamp location long lat and individual*/
 
-main () {
-  //declaring variables
-  cout<<"hi";
-  string line;
-  ifstream myfile ("Baboon_data.csv");
-  vector<string> baboon_names;
-  map<string, vector<vector<string>>> baboon_personal_data;
 
-  if (myfile.is_open()) //opens file
-  {
-
-    while ( getline (myfile,line) ) //assigns each line of the file to the string line.
-    {
-
-      stringstream s_stream(line);// don't quite understand this yet. The documentation was a little confusing, will read further.
-
-      vector<string> result;
-
-      while(s_stream.good()) //goes through each line (while there are lines to go through) and seperates based on the delimeter (the comma)
+class Baboon_Reader {
+public:
+  map<string, vector<tuple<double,double,double>>> readFile(){
+    /*
+    generates a map consisting of baboon names as keys.
+    Each key has a vector listing tuples containing longitude
+    and latitude for each timestamp
+    */
+      cout<<"Creating Map";
+      string line;
+      ifstream myfile ("Baboon_data.csv");
+      vector<string> baboon_names;
+      map<string, vector<tuple<double,double,double>>> baboon_personal_data;
+
+      if (myfile.is_open()) //opens file
       {
-        string substr;
-        getline(s_stream, substr, ',');
-        result.push_back(substr);
-
+        while ( getline (myfile,line) ) //assigns each line of the file to the string line.
+        {
+
+          stringstream s_stream(line);// don't quite understand this yet. The documentation was a little confusing, will read further.
+
+          vector<string> result;
+
+          while(s_stream.good())
+          {
+            /*
+            goes through each line (while there are lines to go through) and seperates based on the delimeter (the comma).
+            Adding them to the result vector.
+            */
+            string substr;
+            getline(s_stream, substr, ',');
+            result.push_back(substr);
+          }
+          if (find(baboon_names.begin(), baboon_names.end(),result[8]) != baboon_names.end()){

```

Comment:
It's good practice to avoid using raw numbers like this, as it can often be hard to know where that "magic number" came from. Try, for example,
```c++
auto const name_index = 8u;
if( find( baboon_names.cbegin(), baboon_names.cend(), result[ name_index ] ) != baboon_names.cend() ) // stuff
```

Reply:
Another helpful formatting recommendation! Will look up the documentation to find out exactly what is happening here